src/routes/user.js

```diff
@@ -0,0 +1,106 @@
+import express from 'express';
+import prisma from '../tools/prisma.js';
+import {
+  ArchivedError,
+  RequestError,
+  AuthenticationError,
+} from '../constants/commonErrors.js';
+
+const router = express.Router();
+
+//welcomes users
+router.get('/', (request, response) => {
+  response.json({ message: 'Welcome!' });
+});
+
+router.get('/archived', () => {
+  throw new ArchivedError('This user has been archived');
+});
+
+router.get('/error', () => {
+  throw new RequestError('Request Error');
+});
+
+router.get('/auth', () => {
+  throw new AuthenticationError('Authentication Error');
+});
+
+//gets a user from id
+router.get('/:id', async (request, response, next) => {
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        id: request.params.id,
+      },
+    });
+    if (!user) {
+      throw new RequestError('User not found with ID');
+    }
+    response.json({ user });
+  } catch (error) {
+    next(error);
+    return;
+  }
+});
+
+//update user by id
+router.patch('/:id', async (request, response, next) => {
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        id: request.params.id,
+      },
+    });
+    if (!user) {
+      throw new RequestError('User not found with ID');
+    }
+    //const updateName =
+    await prisma.user.update({
+      where: {
+        id: request.params.id,
+      },
+      data: {
+        name: request.body.name,
```

Comment:
Our user model currently only has two mutable attributes: name and email. The other attributes in the model should stay constant as it would be illogical for the user to be able to access things like createdAt and updatedAt times.

Now that you point this out, I do have a question. The code works as it is right now, but it's using at very generic response message. 

<img width="1440" alt="Screenshot 2024-06-19 at 4 22 56 PM" src="https://github.com/HireU-App/api/assets/138735926/6b3a9d17-b716-45cc-85ff-32335e785e96">

I'm trying to add a separate message based on what the updated item is, but the update is occurring everytime at `const nameChange` regardless is `request.body.name` exists or not. How can I fix this?

Reply:
I think I got it...